portfolio/src/main/java/com/google/sps/servlets/AddCommentServlet.java

```diff
@@ -20,19 +32,54 @@
   /* Adds the request comment to the datastore */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    System.out.println("add comment servelt");
     UserService userService = UserServiceFactory.getUserService();
     if (!userService.isUserLoggedIn()) return;
 
     String email = userService.getCurrentUser().getEmail();
     String message = request.getParameter("message");
     long timestamp = System.currentTimeMillis();
+    Optional<String> image = getUploadedFileUrl(request);
 
     Entity newComment = new Entity("Comment");
 
     newComment.setProperty("email", email);
     newComment.setProperty("timestamp", timestamp);
     newComment.setProperty("message", message);
+    image.ifPresent(url -> newComment.setProperty("image", url));
 
     DatastoreServiceFactory.getDatastoreService().put(newComment);
   }
+
+  /** Returns the URL to the uploaded file or null if the user didn't upload a file. */
```

Comment:
nit: use `@param` and `@return` JavaDoc block tag for commenting the parameter and return type. 

https://google.github.io/styleguide/javaguide.html#s7.1.3-javadoc-block-tags

Reply:
I don't think it's necessary to add the block tags personally. But you should probably update to reflect the Optional return value.